rocAL/rocAL/source/meta_data/label_reader_file_list.cpp

```diff
@@ -0,0 +1,214 @@
+
+LabelReaderFileList::LabelReaderFileList()
+{
+    _src_dir = nullptr;
+    _entity = nullptr;
+    _sub_dir = nullptr;
+}
+
+void LabelReaderFileList::init(const MetaDataConfig& cfg, pMetaDataBatch meta_data_batch)
+{
+    _file_list_path = cfg.path();
+    _output = meta_data_batch;
+}
+
+bool LabelReaderFileList::exists(const std::string& image_name)
+{
+    return _map_content.find(image_name) != _map_content.end();
+}
+
+void LabelReaderFileList::add(std::string image_name, int label)
+{
+    pMetaData info = std::make_shared<Label>(label);
+    if(exists(image_name))
+    {
+        WRN("Entity with the same name exists")
+        return;
+    }
+    _map_content.insert(pair<std::string, std::shared_ptr<Label>>(image_name, info));
+}
+
+void LabelReaderFileList::print_map_contents()
+{
+    std::cerr << "\nMap contents: \n";
+    for (auto& elem : _map_content) {
+        std::cerr << "Name :\t " << elem.first << "\t ID:  " << elem.second->get_labels()[0] << std::endl;
+    }
+}
+
+void LabelReaderFileList::release()
+{
+    _map_content.clear();
+}
+
+void LabelReaderFileList::release(std::string image_name)
+{
+    if(!exists(image_name))
+    {
+        WRN("ERROR: Given not present in the map" + image_name);
+        return;
+    }
+    _map_content.erase(image_name);
+}
+
+void LabelReaderFileList::lookup(const std::vector<std::string>& image_names)
+{
+    // std::cerr << "\n Printing Map Contents";
+    // print_map_contents();
+    // std::cerr << "\n End of Map Contents";
+    if(image_names.empty())
+    {
+        WRN("No image names passed")
+        return;
+    }
+    if(image_names.size() != (unsigned)_output->size())
+        _output->resize(image_names.size());
+
+    for(unsigned i = 0; i < image_names.size(); i++)
+    {
+        auto image_name = image_names[i];
+        auto it = _map_content.find(image_name);
+        if(_map_content.end() == it)
+            THROW(" Label Reader File List ERROR: Given name not present in the map "+ image_name )
+        _output->get_labels_batch()[i] = it->second->get_labels();
+    }
+}
+
+void LabelReaderFileList::read_all(const std::string& _path)
+{
+    std::string _folder_path = _path;
+    if ((_sub_dir = opendir (_folder_path.c_str())) == nullptr)
+        THROW("ERROR: Failed opening the directory at " + _folder_path);
+
+    std::vector<std::string> entry_name_list;
+    std::string _full_path = _folder_path;
+
+    while((_entity = readdir (_sub_dir)) != nullptr)
+    {
+        std::string entry_name(_entity->d_name);
+        if (strcmp(_entity->d_name, ".") == 0 || strcmp(_entity->d_name, "..") == 0) continue;
+        entry_name_list.push_back(entry_name);
+    }
+    std::sort(entry_name_list.begin(), entry_name_list.end());
+    closedir(_sub_dir);
+    uint label_counter = 0;
+    std::ifstream infile(_file_list_path);
+    std::string line;
+    while (std::getline(infile, line))
+    {
+        std::istringstream iss(line);
```

Comment:
TRy a meaningful name for the istringstream

Reply:
done